### ISSUE TITLE
cmake: pkgconfig files not installed

### ISSUE THREAD

**Issue by jengelh** [grpc/grpc#10727]
In grpc-1.2.3,
```
[    3s] + mkdir -p build
[    3s] + cd build
[    3s] + /usr/bin/cmake /home/abuild/rpmbuild/BUILD/grpc-1.2.3/. '-GUnix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=/usr/lib64 -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -DNDEBUG' '-DCMAKE_CXX_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -DNDEBUG' '-DCMAKE_Fortran_FLAGS=-fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -DNDEBUG' '-DCMAKE_EXE_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_SHARED_LINKER_FLAGS=-Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_SKIP_RPATH:BOOL=ON -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/share/cmake/Modules -DgRPC_PROTOBUF_PROVIDER=package -DgRPC_SSL_PROVIDER=package -DgRPC_ZLIB_PROVIDER=package -DgRPC_USE_PROTO_LITE=BOOL:ON
[...]
+ make install
[...]
+ find /tmp/F|grep pc$
/tmp/F/usr/include/grpc
```
However, the plain top-level Makefile *does* have rules for creating .pc files, but the cmake thing does not‽

**Comment by scinart** [comment]
#12968 is a duplicate of this issue, and this bug still exists in [v1.9.1](https://github.com/grpc/grpc/releases/tag/v1.9.1)!

**Comment by jtattermusch** [comment]
The cmake build scripts are meant mostly for:
- be able to build on Windows (where cmake is our primary build system).
- to be able to integrate well with cmake based projects on other platforms (cmake is not our primary build system on Linux and Mac).

If you are needing the pkgconfig files, why don't you just install using `make install` (not using cmake at all)? The cmake-based install is there mostly to provide a gRPCConfig.cmake so that other cmake based project can include gRPC using find_package().  So I am not sure if duplicating the pkgconfig files duplication makes sense.

**Comment by jengelh** [comment]
>If you are needing the pkgconfig files, why don't you just install using make install (not using cmake at all)?

Plain makefiles are the worst option when it comes to packaging by a Linux distribution. There is no standardization. Everyone is keen on inventing 1. their own variable names (`PREFIX` vs. `prefix`), 2. funny installation paths / not heeding the FHS split between "bin" and "share" and just dumping everything into one directory, 3. mess up with the _basic_ expectations we usually get from build systems like autotools, cmake, (or whatever else there is). Like the complete lack of a `make install` target, or, in case of grpc, I just found a complete lack of DESTDIR support:

```
[   30s] + make install DESTDIR=/home/abuild/rpmbuild/BUILDROOT/grpc-1.9.0-0.x86_64
[   33s] [INSTALL] Installing public C headers
[   33s] install: cannot create directory '/usr/local/include/grpc': Permission denied
```

So no. Whenever there is a build system file, distros will try to use it. We will even write one ourselves if plain makefilery is just way too bulky/complicated/faulty.

**Comment by jtattermusch** [comment]
@jengelh  do you feel like creating a pull request with the desired changes?

**Comment by coryan** [comment]
> If you are needing the pkgconfig files, why don't you just install using make install (not using
> cmake at all)?

In addition to what @jengelh said: because the Makefiles do not install the `gRPCConfig.cmake` file either.  How the library is used (via CMake, via pkg-config, via something else) should not depend on how the library was compiled and linked.

> So I am not sure if duplicating the pkgconfig files installation makes sense.

So don't.  Use CMake for all builds on Windows, Linux and macOS.  Always install the `*.pc` files (even on Windows, some folks use pkgconfig and Makefiles there too), and always install the `*.cmake` files.  You cut one build system to support (yes, I know you use templates to generate them, but still...), and your users get to use the build system that they wanted to.

**Comment by madscientist42** [comment]
> The cmake build scripts are meant mostly for:
> 
> * be able to build on Windows (where cmake is our primary build system).
> * to be able to integrate well with cmake based projects on other platforms (cmake is not our primary build system on Linux and Mac).
> 
> If you are needing the pkgconfig files, why don't you just install using `make install` (not using cmake at
> all)? The cmake-based install is there mostly to provide a gRPCConfig.cmake so that other cmake
> based project can include gRPC using find_package(). So I am not sure if duplicating the pkgconfig 
>files installation makes sense.

1) Not everything uses CMake (some use Autotools...to many's dismay)
2) Straight-up makefiles do _**NOT**_ work worth jack on cross-compilation.

I'd rather you'd do this "right" in the first place.  I can't get your CMake config files you generate to properly pick the right cross-compilation (You do **NOT** honor all the all of the needed things from a tools file or I'd have my problem fixed already...  You presume your pathing for libraries, etc. is from the same as the tools...which breaks all to HELL on a Yocto setup for follow-on binaries or an SDK sandbox- because your tools are in an X86 environment usually, and the other's in an ARM, MIPS, or RISC-V target sysroot...)  

If you did  both, I'd have had the ability to simply resorted to pkgconfig configuration which would've honored my target sysroot in the follow on projects.  Broken, but usable.  As it stands, I'm now trying to figure out how to un-pooch what you guys've done to me.

**Comment by madscientist42** [comment]
(As an aside, the CMake Config mode package finders are broken quite horribly for **_ANY_** cross-compilation contexts...  You **_DO NOT_** go and presume ${CMAKE_CURRENT_LIST_FILE} is your path for everything...)

**Comment by jtattermusch** [comment]
I believe this is still an issue.

**Comment by KyleFromKitware** [comment]
@jtattermusch Since this has been fixed by #20211, can it be moved to the "Done" category?

**Comment by jtattermusch** [comment]
@KyleFromKitware  moved.
